torchtext/datasets/amazonreviewpolarity.py

```diff
@@ -25,20 +27,24 @@
     'test': f'{os.sep}'.join(['amazon_review_polarity_csv', 'test.csv']),
 }
 
-_EXTRACTED_FILES_MD5 = {
-    'train': "520937107c39a2d1d1f66cd410e9ed9e",
-    'test': "f4c8bded2ecbde5f996b675db6228f16"
-}
 
 DATASET_NAME = "AmazonReviewPolarity"
 
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=2)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def AmazonReviewPolarity(root, split):
-    path = _download_extract_validate(root, URL, MD5, os.path.join(root, _PATH), os.path.join(root, _EXTRACTED_FILES[split]),
-                                      _EXTRACTED_FILES_MD5[split], hash_type="md5")
-    logging.info('Creating {} data'.format(split))
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def AmazonReviewPolarity(root: str, split: Union[Tuple[str], str]):
+    # TODO Remove this after removing conditional dependency
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL])
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+    )
+    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, mode="b")
+    extracted_files = cache_dp.read_from_tar()
+    filter_extracted_files = extracted_files.filter(lambda x: split in x[0])
```

Comment:
@parmeet I didn't catch this earlier, but I strongly recommend updating this logic to include the extension of the split (i.e. `.csv` or `.tsv`). This caused bugs in the SST2 implemenation in the past. Look at https://github.com/pytorch/text/pull/1444 for reference

Reply:
An alternative would be to do `lambda x: _EXTRACTED_FILES[split] in x[0]` which is what @erip is doing in https://github.com/pytorch/text/pull/1499

Reply:
Thanks @Nayef211 for catching this. Yes, I remember this breakage earlier, and completely forgot to take care of this. Will do PR to fix it.